src/hooks/Guilds/guild/useProposalCalls.ts

```diff
@@ -34,27 +37,50 @@ const useProposalCalls = (guildId: string, proposalId: string) => {
   } = proposal || {};
 
   const totalOptionsNum = totalVotes?.length || 0;
+  const displayableOptionsNum = totalOptionsNum - 1;
+  const nonApprovalOrEmptyCalls = useMemo(
+    () => dataArray?.filter(data => !isApprovalData(data) && !isZeroHash(data)),
+    [dataArray]
+  );
+
   const callsPerOption = totalOptionsNum
-    ? toArray?.length / totalOptionsNum
+    ? nonApprovalOrEmptyCalls.length / displayableOptionsNum
     : 0;
   const optionLabels = metadata?.voteOptions;
 
   const calls: Call[] = useMemo(() => {
-    return toArray?.map((to, index) => ({
+    const buildCall = idx => ({
       from: guildId,
-      to: to,
-      data: dataArray[index],
-      value: valuesArray[index],
-    }));
-  }, [guildId, dataArray, valuesArray, toArray]);
+      to: toArray[idx],
+      data: dataArray[idx],
+      value: valuesArray[idx],
+    });
+    return dataArray
+      ?.map((dataValue, index) => {
+        const call = buildCall(index);
+        if (isApprovalData(dataArray[index - 1])) {
+          return {
+            ...call,
+            // We asume that if previous call was an approval, then current one is the one that is being approved
+            // So passing nested approval call and remove it from the calls array
+            approvalCall: buildCall(index - 1),
+          };
+        }
+
+        if (isApprovalCall(call) || dataValue === ZERO_HASH) return null;

```

Comment:
💡 Could reuse the `isZeroHash`-function from above?

Reply:
Yes, we can! Done.